app/models/user.rb

```diff
@@ -0,0 +1,17 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z0-9\-.]+\.[a-z]+\z/i.freeze
+
+  has_secure_password
+
+  validates :name, presence: true
+  validates :email, presence: true, length: {maximun: Settings.email.max, minimum: Settings.email.min},
+  uniqueness: true, format: {with: VALID_EMAIL_REGEX}
+
+  before_save :downcase_email
+
+  private
+
+  def downcase_email
+    email.downcase!
```

Comment:
`email.downcase!` và `email.downcase` khác gì nhau

Reply:
Dạ chị, theo em hiểu thì, downcase và downcase! khác nhau ở chỗ downcase! sẽ làm thay đổi giá trị của biến còn downcase thì không. Ngoài ra, nếu khi ta downcase! nhưng giá trị mới giống giá trị cũ (tức là biến không có ký tự in hoa để thay đổi) thì downcase! sẽ trả về nill, còn downcase thì trả về giá trị của biến đó.